From 6b95e4257ff78276898c24af0f676c52c02d9ca2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Sat, 1 Feb 2020 09:14:46 +0100
Subject: [PATCH] install: Raise an error to avoid breaking python-exec

Running pip without --target, --prefix, --root, or --user will result
in packages being installed systemwide. This has a tendency to break
python-exec if setuptools gets installed or upgraded.
---
 src/pip/_internal/commands/install.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/pip/_internal/commands/install.py b/src/pip/_internal/commands/install.py
index b16b8e3..2dbfa60 100644
--- a/src/pip/_internal/commands/install.py
+++ b/src/pip/_internal/commands/install.py
@@ -302,6 +302,9 @@ class InstallCommand(RequirementCommand):
             upgrade_strategy = options.upgrade_strategy
 
         cmdoptions.check_build_constraints(options)
+        if not options.use_user_site and not options.target_dir and not options.root_path and not options.prefix_path and not os.getenv('GENTOO_PIP_TESTING'):
+            raise CommandError("(Gentoo) Please run pip with the --user option to avoid breaking python-exec")
+
         cmdoptions.check_dist_restriction(options, check_target=True)
 
         logger.verbose("Using %s", get_pip_version())
-- 
2.49.0.9.gd50a5e8939.dirty

